### PR TITLE
[4.0] Make sure the function matches the parent return type

### PR DESCRIPTION
**Pull request by roland-d**
### Summary of Changes
This fixes a fatal error when accessing the site

`Fatal error: Declaration of Joomla\CMS\Session\Storage\JoomlaStorage::setOptions(array $options) must be compatible with Joomla\Session\Storage\NativeStorage::setOptions(array $options): Joomla\Session\Storage\NativeStorage in libraries/src/Session/Storage/JoomlaStorage.php on line 23`

### Testing Instructions
1. Checkout the current Joomla 4 branch
2. Open the site and you get this fatal error:
```
Fatal error: Declaration of Joomla\CMS\Session\Storage\JoomlaStorage::setOptions(array $options) must be compatible with Joomla\Session\Storage\NativeStorage::setOptions(array $options): Joomla\Session\Storage\NativeStorage in libraries/src/Session/Storage/JoomlaStorage.php on line 23
```
3. Apply this pull request
4. Run `composer update`
5. Reload the site
6. Site now loads

### Expected result
The site loads

### Actual result
The site doesn't load

### Documentation Changes Required
None

Pinging @wilsonge

### CODE REVIEW

File: libraries/src/Session/Session.php [hunk 1]
```diff
@@ -230,10 +230,10 @@ public function clear()
 					$name = $args[1] . '.' . $name;
 				}
 
-				return $this->remove($name);
+				$this->remove($name);
```
Comment (wilsonge) [inline libraries/src/Session/Session.php]: this is a b/c break. needs reverting the void typehint in the framework i'm afraid :).

the entire point of the compat code is to return here

**Comment by mbabker** [comment]
Also for clarity, on the `Fatal error: A void function must not return a value in /usr/local/var/www/joomlacms/libraries/src/Session/Session.php on line 233` error...

Everything that's been typehinted is either new API for CMS 4.0 or Framework 2.0, or exists in a non-extendable scope (final class, private method, etc.).  Anything that already exists in a stable CMS 3.x or Framework 1.x shouldn't be typehinted because of B/C breaks in changing signatures, unless the change was explicitly vetted (and there are some, but they're the exception to the rule).  That one slipped in because we lost the `@since 1.0` tag on the parent method in the Framework repo (it had `@since __DEPLOY_VERSION__` so I figured it was new when going through things).  So if you start running into type issues, first look for a B/C break then revert that break (unless explicitly vetted and OK'd as such) before trying to adapt code to the changed signature.

**Comment by richard67** [comment]
@roland-d Is it ready for testing?

**Comment by roland-d** [comment]
@mbabker Thanks for the input

@richard67 It should be ready now, I cleaned up the test instructions.

**Comment by wilsonge** [comment]
Tested this locally and works. Thanks!

**Comment by richard67** [comment]
@wilsonge Bad boy 👅 . I was so keen on testing it, and now you merged already :smile:

**Comment by wilsonge** [comment]
sorrrrrrry :/

**Pull request merged**